README.md

```diff
@@ -13,3 +13,76 @@
+  
+  Asymptotic analysis doesn't account for constants, and bases its performance
+  exclusively on the dominant term. For example an algorithm with a complexity
+  of $4n^2+n$ gets baked down to $O(n^2)$. Sure, as our data set approaches
+  an infinite quantity of entries, those constants and lower order terms mean
+  very little, but in the real world those eccentricities still mean something.  
+  
+  Asymptotic analysis doesn't account for what language a program is written in.
+  Broadly speaking different programming languages are capable of performing the
+  same tasks; however, they perform these tasks with different efficacies. A
+  significant distinction is compiled vs interpreted languages. Compiled
+  languages typically offer better performances, and even amongst all compiled
+  languages, each will have different performance capabilities. Though
+  insignificant with small data sets, the same algorithm sorting the same data
+  in C vs Python, when sorting large data sets, could result in dramatically
+  different real world performances.  
+
+## Question 2
+
+  A BST has a asymptotic complexity of $O(log_{2} n)$. Knowing that $T(n)$ for
+  $n=1000$ takes 5 seconds, we can calculate a ratio, $c$, by which we can
+  calculate $T(10000)$.  
+  $c \cdot \log_{2} 1000 = 5$  
+  $c = \frac{5}{\log_{2} 1000}$  
+  Using that ratio, $c$  
+  $T(10000) = \frac{5}{\log_{2} 1000} \cdot \log(10000)$  
+  $T(10000) \approx 0.5017 \cdot 13.2877$  
+  $T(10000) \approx 6.6664$  
+  Thus, I'd guess T(10000) would take 6.6664 seconds.  
+
+## Question 3
+
+  I'm combining two of my previous answers, as they were ostensibly the same
+  main idea.  
+
+  Hardware limitations, like a slow CPU, or limited RAM could allow for small
+  data sets to be processed relatively quickly, but larger data sets that
+  don't fully fit into your CPU's cache, or main memory could lose a lot of
+  time loading, and being moved around the file system.  
+
+  Perhaps my implimentation of a BST was deeply flawed, and that combined with
+  a really weird data set is resulting in an extremely off balance tree,
+  causing the program to need to search far deeper than should be necessary.
+  Or my algorithm to find the element was poorly implimented, and my code
+  checks every single node, even after the element is found, 'cause I didn't
+  include a break statement. I guess poor coding would be the general idea here.  
+
+  Another thing that could cause unexpected differences in performances with
+  larger data sets is what I suspect is taking my in place merge sort program
+  such an absurd amount of time to finish testing, 71 minutes as of the time of
+  writing, many memory operations. Merge sort isn't BST, but the point is
+  operations that take a negligible amount of time for a small data set have
+  the potential to scale up to a massive performance bottleneck for larger
+  data sets.  
```

Comment:
But that would only be a constant factor, wouldn't it? I don't really understand this reason.